src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for constructing Vietoris-Rips complexes and computing the persistence diagrams from them.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        threshold=float('inf'),
+        input_type='point cloud',
+        num_collapses=1,
+        homology_coeff_field=11,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            threshold (float): Rips maximal edge length value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
+                inputs are lower triangular distance matrix (can be full square, but the upper part of the distance
+                matrix will not be considered). Default is 'point cloud'.
+            num_collapses (int): Specify the number of :func:`~gudhi.SimplexTree.collapse_edges` iterations to perform
+                on the SimplexTree. Default value is 1 (a relatively good enough number of iterations).
+            homology_coeff_field (int): The homology coefficient field. Must be a prime number. Default value is 11.
+            expand_extra_dimension (bool): :func:`~gudhi.SimplexTree.expansion` is performed at
+                `max(homology_dimensions) + 1` if true, and `max(homology_dimensions)` otherwise. Default is true.
```

Comment:
I am thinking of removing this option from RipsPersistence. While the max_dim thing makes sense for a general simplicial complex, I don't think it (expand_extra_dimension=False) makes much sense for a flag complex. In any case, it is incompatible with edge collapse (and likely ripser), which assumes a true flag complex, not a truncated one (I didn't check, but I think already with 3 points (acute triangle), expand_extra_dimension=False and homology_dimensions=1, the diagram should contain 1 point without collapse, and 0 with collapse).

Reply:
On [81ebe84](https://github.com/GUDHI/gudhi-devel/pull/691/commits/81ebe849565bdc4782be59577e6cf76f3b197bc2), I removed useless expand_extra_dimension argument and modify persistence_dim_max accordingly